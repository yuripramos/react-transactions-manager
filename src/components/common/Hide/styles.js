import styled, { css } from "styled-components";
import { media } from "../../../styles/tools";

export const HideStyle = styled.div`
  display: none;

  ${({ below }) =>
    below &&
    media[below](css`
      display: block;
    `)};

  ${({ above }) =>
    above &&
    css`
      display: block;

      ${media[above](css`
        display: none;
      `)};
    `};
`;
