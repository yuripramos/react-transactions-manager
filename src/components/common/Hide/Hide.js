import React from "react";
import { string } from "prop-types";
import { HideStyle } from "./styles";

function Hide({ children, below, above }) {
  return (
    <HideStyle below={below} above={above}>
      {children}
    </HideStyle>
  );
}

Hide.displayName = "Hide";

Hide.defaultProps = {
  below: null,
  above: null
};

Hide.propTypes = {
  below: string,
  above: string
};

export default Hide;
