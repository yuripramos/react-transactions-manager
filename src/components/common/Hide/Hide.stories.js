import React from "react";
import { storiesOf } from "@storybook/react"; // eslint-disable-line
import { withInfo } from "@storybook/addon-info"; // eslint-disable-line

import Hide from "./Hide";

storiesOf("common/Hide", module)
  .add(
    "hide above above sm",
    withInfo()(() => <Hide above="sm">Hide title</Hide>)
  )
  .add(
    "hide above below sm",
    withInfo()(() => <Hide below="sm">Hide title</Hide>)
  )
  .add(
    "hide above above md",
    withInfo()(() => <Hide above="md">Hide title</Hide>)
  )
  .add(
    "hide above below md",
    withInfo()(() => <Hide below="md">Hide title</Hide>)
  )
  .add(
    "hide above above lg",
    withInfo()(() => <Hide above="lg">Hide title</Hide>)
  )
  .add(
    "hide above below lg",
    withInfo()(() => <Hide below="lg">Hide title</Hide>)
  );
