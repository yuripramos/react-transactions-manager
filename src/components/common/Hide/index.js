import Hide from "./Hide";

export default Hide;
