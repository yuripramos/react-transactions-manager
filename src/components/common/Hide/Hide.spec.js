import React from "react";
import { render } from "enzyme";

import Hide from "./Hide";

describe("Hide component", () => {
  it("should match snapshot changing class", () => {
    expect(render(<Hide>Text</Hide>)).toMatchSnapshot();
  });

  it("should match snapshot changing class", () => {
    expect(render(<Hide below="md">Text</Hide>)).toMatchSnapshot();
  });

  it("should match snapshot changing class", () => {
    expect(render(<Hide above="md">Text</Hide>)).toMatchSnapshot();
  });
});
