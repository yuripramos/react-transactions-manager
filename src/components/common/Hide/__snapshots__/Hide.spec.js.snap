// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hide component should match snapshot changing class 1`] = `
<div
  class="sc-bdVaJa kxLTFn"
>
  Text
</div>
`;

exports[`Hide component should match snapshot changing class 2`] = `
<div
  class="sc-bdVaJa keaLpg"
>
  Text
</div>
`;

exports[`Hide component should match snapshot changing class 3`] = `
<div
  class="sc-bdVaJa cVegwE"
>
  Text
</div>
`;
